projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1943 @@
+
+ @param args - pointer to the arguments on the command line.
+
+ @return none
+ **/
+void RunPulse(uint8_t *args)
+{
+	eSensorResult = pGasSensor->openWithAddr(uiDefaultAddress);
+
+	flushBuff(gBuff, sizeof(gBuff));
+	eSensorResult = pGasSensor->Read200RCal(gBuff);
+	rcal = gBuff[0];
+
+	flushBuff(TXbuff, sizeof(TXbuff));
+	eSensorResult = pGasSensor->RunPulseTest(uiPulseAmpl, uiPulseDur);
+	if (eSensorResult != SENSOR_ERROR_NONE) {
+		UART_TX("ERROR!" _EOS);
+	} else {
+		sprintf((char*) TXbuff, "Pulse test started%s", _EOS);
+		UART_TX((const char*) TXbuff);
+	}
+	pGasSensor->close();
+}
+
+/**
+ @brief Read results of chronoamperometry pulse test
+
+ @param args - pointer to the arguments on the command line.
+
+ @return none
+ **/
+void ReadPulse(uint8_t *args)
+{
+	uint16_t value = 0;
+	eSensorResult = pGasSensor->openWithAddr(uiDefaultAddress);
+	flushBuff(TXbuff, sizeof(TXbuff));
+	eSensorResult = pGasSensor->ReadPulseTestResults(&pulseResultBuffer[0],
+			uiPulseAmpl, uiPulseDur);
+	if (eSensorResult != SENSOR_ERROR_NONE) {
+		UART_TX("ERROR!" _EOS);
+		pGasSensor->close();
+		return;
+	}
+
+	UART_TX("time [msec], current [uA]" _EOS);
+	uint16_t valToSend = (((uiPulseDur + 10) * 1000) / 110);
+	uint16_t cnt = 0;
+	uint16_t cnt2 = 1;
+	while (cnt < valToSend) {
+		flushBuff(TXbuff, sizeof(TXbuff));
+		value = ((pulseResultBuffer[cnt2] << 8)
+			 | pulseResultBuffer[cnt2 + 1]);
+		sprintf((char*) TXbuff, "%d, %.2f%s", ((cnt) * 110),
+			((32768.0 - value) / 65535)
+			* ((1835 * 2 * 1000) / rcal),
+			_EOS);
+		UART_TX((const char*) TXbuff);
+		cnt++;
+		cnt2 += 2;
+	}
+
+	pGasSensor->close();
+}
+
+/**
+ @brief Set step amplitude for pulse test
+
+ @param args - pointer to the arguments on the command line.
+ amplitude: [mV] - Amplitude of the pulse in mV (typically 1mV)
+
+ @return none
+ **/
+void SetPulseAmpl(uint8_t *args)
+{
+	uint8_t *p = args;
+	char arg[17];
+	uint8_t i;
+	uint8_t digitcount = 0;
+
+	/* Check if this function gets an argument */
+	while (*(p = FindArgv(p)) != '\0') {
+		/* Save channel parameter */
+		GetArgv(arg, p);
+	}
+
+	//Check if arg is numeric
+	for (i = 0; arg[i] != '\0'; i++) {
+		// check for decimal digits
+		if (isdigit(arg[i]) != 0)
+			digitcount++;
+	}
+	if (digitcount == i) { //All characters are numeric
+		uiPulseAmpl = (uint8_t) strtol(arg, (char **) NULL,
+					       10); //convert string to number
+	} else {
+		UART_TX("Invalid entry!" _EOS);
+		return;
+	}
+
+	if (uiPulseAmpl >= 3 || uiPulseAmpl < 1) {
+		uiPulseAmpl = 1;
+		UART_TX("Out of range! Set to default = 1 mV" _EOS);
+		return;
+	}
+
+	flushBuff(gBuff, sizeof(gBuff));
+	sprintf((char *) gBuff, "Pulse amplitude set to %d mV%s", uiPulseAmpl,
+		_EOS);
+	UART_TX((const char*) gBuff);
+}
+
+/**
+ @brief Set duration for CAPA test
+
+ @param args - pointer to the arguments on the command line.
+ duration: [milliseconds] - Duration of the pulse (keep <200 ms)
+
+ @return none
+ **/
+void SetPulseDuration(uint8_t *args)
```

Comment:
All of this functions can go into a cn0428/9.c file. With the name changed to: 

CN0428_SetPulseDuration(). This will inform the user that these are functions specific to this CN.